### PR TITLE
Replace location of apoydence/eachers to poy/eachers

### PR DESCRIPTION
**Pull request by ph**
The github repository at `apoydence/eachers` doesn't exist anymore, the
new location of the code is now at `poy/eachers`. The dependency is an
indirect dependency included by https://github.com/cloudfoundry/noaa/blob/5ce49363dfa666c00fc716cb5f8c2f99462b41ef/consumer/token_refresher_test.go#L11
which is used for our cloudfoundry code.

Note the eachers dependency s used while testing the nooa package, but as of today `go mod` is including
these dependency.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

**Comment by ruflin** [comment]
Change itself LGTM. I wonder if there is anything we should contribute back to the package itself to get this cleaned up?

File: go.mod [hunk 1]
```diff
@@ -296,3 +296,5 @@ replace (
 	github.com/insomniacslk/dhcp => github.com/elastic/dhcp v0.0.0-20200227161230-57ec251c7eb3 // indirect
 	github.com/tonistiigi/fifo => github.com/containerd/fifo v0.0.0-20190816180239-bda0ff6ed73c
 )
+
+replace github.com/apoydence/eachers => github.com/poy/eachers v0.0.0-20181020210610-23942921fe77
```
Comment (ruflin) [inline go.mod:300]: Does go.mod support comments? If yes, likely worth to add on here.

Comment (ph) [reply go.mod:300]: @ruflin I am not sure that comment would stay, it's a mostly generated file, I think everything in that files is subject to be regenerated by a tool or `go mod tidy`

Concerning contributing changes upstream, It's something we could do but before I think we should look at our usage of cloudfoundry's noaa, last changes into that library is 5 years ago and we included it 2 years ago. Which is an eternity in code life.

@jsoriano I believe you have implemented the cloudfoundry support a few years ago and you make use of noaa's client, I wonder if this is still the valid way to do in 2022?

Comment (jsoriano) [reply go.mod:300]: > Concerning contributing changes upstream, It's something we could do but before I think we should look at our usage of cloudfoundry's noaa, last changes into that library is 5 years ago and we included it 2 years ago. Which is an eternity in code life.
>
> @jsoriano I believe you have implemented the cloudfoundry support a few years ago and you make use of noaa's client, I wonder if this is still the valid way to do in 2022?

Yeah, this noaa library is used to consume from the v1 API version of the Cloudfoundry firehose, this version has been deprecated for years, but is (or used to be) more reliable in most deployments than the current v2 API, this is why it is still used, and why we implemented support for this version, and made it the default, after having support for v2.

We'd have to check if the situation has changed during this time, but in the meantime I think that we can continue using noaa as is.

**Review by jsoriano — APPROVED** [review]
Agree with adding the replace.

We can have a follow up to re-evaluate the situation regarding the different options to collect events from the firehose.

File: go.mod (hunk 1 above)
[the remark this answers is not in the document]
Comment (ph) [reply go.mod:300]: good catch, I guess the `go mod replace` is not super intelligent to merge these.

File: go.mod [hunk 2]
```diff
@@ -294,5 +295,5 @@ replace (
 	github.com/golang/glog => github.com/elastic/glog v1.0.1-0.20210831205241-7d8b5c89dfc4
 	github.com/google/gopacket => github.com/elastic/gopacket v1.1.20-0.20211202005954-d412fca7f83a
 	github.com/insomniacslk/dhcp => github.com/elastic/dhcp v0.0.0-20200227161230-57ec251c7eb3 // indirect
-	github.com/tonistiigi/fifo => github.com/containerd/fifo v0.0.0-20190816180239-bda0ff6ed73c
+	github.com/tonistiigi/fifo => github.com/containerd/fifo v0.0.0-20190816180239-bda0ff6ed73c //indirect, see https://github.com/elastic/beats/pull/29780 for details.
```
Comment (jsoriano) [inline go.mod:298]: I think this comment doesn't belong to this line :)

**Review by jsoriano — APPROVED** [review]

**Pull request merged**